tests/unit/test_lib_per_app_provides.py

```diff
@@ -72,7 +74,7 @@ def test_ingress_app_provider_relate_provide(
     relation = harness.model.get_relation("ingress", relation_id)
     assert provider.is_ready(relation)
 
-    provider.publish_url(relation, "foo.com")
+    provider.publish_url(relation, "baz://foo.com")
 
     ingress = harness.get_relation_data(relation_id, "test-provider")["ingress"]
-    assert yaml.safe_load(ingress) == {"url": "foo.com"}
+    assert yaml.safe_load(ingress) == {"url": "baz://foo.com"}
```

Comment:
This is surprising: the enum (that I think we should drop) allows this?

Reply:
the library doesn't validate the advertised url
What traefik decides to publish gets put in relation data as-is. maybe we should? but that's for another feature :)